main.go

```diff
@@ -110,7 +110,68 @@ func layout(g *gocui.Gui) error {
 	}
 	return nil
 }
+func scrollUP(v *gocui.View) error {
+	scrollView(v, -1)
+	return nil
+}
+func scrollDOWN(v *gocui.View) error {
+	scrollView(v, 1)
+	return nil
+}
+func scrollView(v *gocui.View, delta int) error {
+	if v != nil {
+		_, y := v.Size()
+		ox, oy := v.Origin()
+		if oy+delta > strings.Count(v.ViewBuffer(), "\n")-y {
+			v.Autoscroll = true
+		} else {
+			v.Autoscroll = false
+			if err := v.SetOrigin(ox, oy+delta); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
```

Comment:
What happens if scrolling goes further than the amount of messages given by populateChat()?

Reply:
it can't - so this grabs the entire view size (which is dictated by `populateChat()`) and scrolls.  it stops at the top end where the `oy` hits the top.  At the bottom end (where `oy+delta > #lines`) it restores auto scrolling and you just continue following the chat as normal.

Reply:
clarification - "It has hard stops on both sides." 
gocui won't let you negative origin, so thats the hard stop at the top.
At the bottom when the index passes the number of lines in the view size, it restores autoscrolling instead of scrolling past the bottom.

Reply:
What sets #lines ? 

Will this only allow the user to scroll through messages that have arrived since they've joined the buffer- and nothing before?

Reply:
On line 125, `strings.Count(v.ViewBuffer(), “\n”)` counts the number of lines currently loaded into the view buffer. 

We could consider a feature enhancement to view more when you hit the top, but that’s more involved. 

In my experience the existing buffer goes back quite a ways (several hours in the kbtui chat)

Reply:
Huh.. I wonder if that was a change, or a mathematical error on my part xD Originally I was only pulling as many messages as there were lines in the Chat view.